src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Allows for the comparison of two <see cref="ISymbol"/> instance based on certain criteria TODO
+    /// </summary>
+    public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol>
+    {
+        /// <summary>
+        /// Compares two <see cref="ISymbol"/> instances based on the default comparison rules, equivalent to calling <see cref="IEquatable{ISymbol}.Equals(ISymbol)"/>
+        /// </summary>
+        public static readonly SymbolEqualityComparer Default = new SymbolEqualityComparer(TypeCompareKind.AllNullableIgnoreOptions);
+
+        /// <summary>
+        /// Compares  two <see cref="ISymbol"/> instances, considering their nullability
+        /// </summary>
+        public static readonly SymbolEqualityComparer IncludeNullability = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything2); //TODO: should this be explicitly *not* compare everything
+
+        // Internal only comparisons:
+        internal static readonly SymbolEqualityComparer ConsiderEverything = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything);
```

Comment:
>ConsiderEverything [](start = 56, length = 18)

Are these `internal` versions used by the public API? If not, could these be defined on a comparer of `Symbol` or `TypeSymbol` instead?